src/main/java/com/dfsek/terra/structure/Structure.java

```diff
@@ -71,18 +71,18 @@ public Structure(@NotNull Location l1, @NotNull Location l2, @NotNull String id)
                                     if(s.getLine(1).equals("[CENTER]")) {
                                         centerX = x;
                                         centerZ = z;
-                                    } else if(s.getLine(1).startsWith("[SPAWN=") && s.getLine(1).endsWith("]")) {
+                                    } else if(s.getLine(1).startsWith("[SPAWN=") && !s.getLine(1).isEmpty() && s.getLine(1).charAt(s.getLine(1).length() - 1) == ']') {
```

Comment:
This is not a performance-sensitive area, and this decreases readability and makes code smellier.